hs/Database/Tables.hs

```diff
@@ -183,29 +162,19 @@ data Course =
 instance ToJSON Course
 instance ToJSON Session
 
-instance ToJSON Lecture where
-  toJSON (Lecture lectureExtra lectureSection lectureCap lectureTimeStr lectureTime lectureInstructor lectureEnrol lectureWait)
-          = object ["extra" .= lectureExtra,
-                    "section" .= lectureSection,
-                    "cap" .= lectureCap,
-                    "time_str" .= lectureTimeStr,
-                    "time" .= map convertTimeToString lectureTime,
-                    "instructor" .= lectureInstructor,
-                    "enrol" .= lectureEnrol,
-                    "wait" .= lectureWait
-                   ]
-
-instance ToJSON Tutorial where
-  toJSON (Tutorial Nothing tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
-  toJSON (Tutorial (Just value) tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON value, toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
+instance ToJSON Time where
+    toJSON (time) =
+        toJSON $ convertTimeToString time
+
+-- instance FromJSON required so that tables can be parsed into JSON,
+-- not necessary otherwise.
+instance FromJSON Time
```

Comment:
1) Yes, neither the day of the week nor the time should be represented by Doubles. This is a separate issue, however.
2) It's not clear to me what you mean by "parsing level". If you're suggesting that we use the default (generic) FromJSON *and* ToJSON, and then parse the json in the Javascript when making the sections, that's fine with me, and probably better, too.